File: clients/android/NewsBlur/src/com/newsblur/fragment/ProfileActivitiesFragment.java
package com.newsblur.fragment;

import android.content.Context;

import com.newsblur.domain.ActivityDetails;
import com.newsblur.domain.UserDetails;
import com.newsblur.network.domain.ActivitiesResponse;
import com.newsblur.view.ActivitiesAdapter;
import com.newsblur.view.ActivityDetailsAdapter;

/**
 * Created by mark on 15/06/15.
 */
public class ProfileActivitiesFragment extends ProfileActivityDetailsFragment {

    @Override
    protected ActivityDetailsAdapter createAdapter(Context context, UserDetails user) {
        return new ActivitiesAdapter(context, user);
    }

    @Override
    protected ActivityDetails[] loadActivityDetails(String id, int pageNumber) {
        ActivitiesResponse activitiesResponse = apiManager.getActivities(id, pageNumber);
        if (activitiesResponse != null) {
            return activitiesResponse.activities;
        } else {
            return new ActivityDetails[0];
        }
    }
}

File: clients/android/NewsBlur/app/src/main/java/com/newsblur/fragment/ProfileActivitiesFragment.kt
package com.newsblur.fragment

import android.content.Context
import com.newsblur.domain.ActivityDetails
import com.newsblur.domain.UserDetails
import com.newsblur.util.ImageLoader
import com.newsblur.view.ActivitiesAdapter
import com.newsblur.view.ActivityDetailsAdapter

/**
 * Created by mark on 15/06/15.
 */
class ProfileActivitiesFragment : ProfileActivityDetailsFragment() {
    override fun createAdapter(
        context: Context?,
        user: UserDetails?,
        iconLoader: ImageLoader,
    ): ActivityDetailsAdapter? = ActivitiesAdapter(context, user, iconLoader)

    override suspend fun loadActivityDetails(
        id: String,
        pageNumber: Int,
    ): Array<ActivityDetails>? {
        val activitiesResponse = userApi.getActivities(id, pageNumber)
        return activitiesResponse?.activities
    }
}
